File: backends/kotlin-ktor/src/main/kotlin/com/authorization/usecase/auth/Dto.kt
/*
 * 認証ユースケース DTO モジュール。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 */
package com.authorization.usecase.auth

/**
 * ログイン DTO です。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 */
data class LoginDto(
    val provider:        Int,
    val providerId:      String,
    val name:            String,
    val email:           String,
    val avatar:          String?,
    val invitationToken: String? = null,
)

File: backends/ruby-rails/app/usecase/auth/dto.rb
# frozen_string_literal: true
#
# 認証ユースケースの DTO を定義するモジュール。
#
# @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>

module UseCase
  module Auth
    # ログイン時に渡す DTO です。
    LoginDto = Struct.new(:provider, :provider_id, :name, :email, :avatar, keyword_init: true)
  end
end
